plugin/pkg/shoot/validator/admission.go

```diff
@@ -805,21 +824,55 @@ func hasDomainIntersection(domainA, domainB string) bool {
 	return strings.HasSuffix(long, short)
 }
 
-func validateKubernetesVersionConstraints(constraints []string, version, oldVersion string) (bool, []string) {
-	if version == oldVersion {
-		return true, nil
+func validateKubernetesVersionConstraints(constraints []string, shootVersion, oldShootVersion string) (bool, []string, *semver.Version) {
+	if shootVersion == oldShootVersion {
+		return true, nil, nil
+	}
+
+	shootVersionSplit := strings.Split(shootVersion, ".")
+	var (
+		shootVersionMajor, shootVersionMinor int64
+		getLatestPatchVersion                bool
+	)
+	if len(shootVersionSplit) == 2 {
+		// add a fake patch version to avoid manual parsing
+		fakeShootVersion := shootVersion + ".0"
+		version, err := semver.NewVersion(fakeShootVersion)
+		if err == nil {
```

Comment:
what is happening when `err != nil` ? I know it is very unlikely to happen, but it is not impossible.
I would invert this condition and report the error.

Reply:
Thanks for the review!  
What we are doing here is to check if the major and minor version provided completed with a dummy patch version are valid. If so, we assume that we want to compute a latest patch version for that minor version.
If there is an error (what will happen any time major + minor are not valid semver like "12.a2") we still want to return all the possible valid values and display them to the user. We collect them in the following loop. That is the reason why we purposefully not return the error here.